controllers/bestie_controller_metrics.go

```diff
@@ -28,15 +28,15 @@ var (
 			Help: "Number of successful bestie application upgrades processed",
 		},
 	)
-	applicationUpgradeFailuresCounter = prometheus.NewCounter(
-		prometheus.CounterOpts{
-			Name: "bestie_upgrade_failures_counter",
-			Help: "Number of failed bestie application upgrades",
+	applicationUpgradeFailure = prometheus.NewGauge(
```

Comment:
may be should be `applicationUpgradeFailureState` or `applicationUpgradeFailureGauge`

Reply:
I agree